application/holder/src/env/os/platform.ts

```diff
@@ -28,13 +29,14 @@ export function getPlatform(win32Only = false): Platform {
             break;
         case Platform.openbsd:
             return Platform.linux;
-        case Platform.darwin:
-            if (os.arch() === 'arm64') {
+        case Platform.darwin: {
+            const result = execSync('sysctl -n machdep.cpu.brand_string').toString();
```

Comment:
`execSync` from [docs](https://nodejs.org/api/child_process.html#child_processexecsynccommand-options) : 

> If the process times out or has a non-zero exit code, this method will throw. 

Very dangerous because:

- it will be called at a very start of an application
- if some error - we will get `throw` during starting, it means an application probably will not start at all
- this is sync method, if some "issues" on a host (magic update of OS for example), we have a risk to wait here forever

I do not mind using something a platform specific, but here we cannot spawn external programs with potentially unpredictable result. It should be wrapped somehow or we should find NodeJS API to do that. Or maybe rust method, which can be called from here

Reply:
Hi Dmitry, updated the code to include the changes you suggested. Let me know if anything else is required.